src/sdam/monitor.rs

```diff
@@ -102,13 +117,19 @@ impl Monitor {
         while self.is_alive() {
             let check_succeeded = self.check_server().await;
 
+            if self.topology_version.is_some() && self.allow_streaming {
```

Comment:
The spec changed from allowing drivers to use a dedicated RTT monitor even when polling to requiring that one not be used when polling, so starting up that task had to be deferred to the point where we know whether polling is happening or not.